core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -34,7 +34,7 @@
  * using a {@link StaticTableOperations}. This way no Catalog related calls are needed when reading the table data after
  * deserialization.
  */
-abstract class BaseMetadataTable implements Table, Serializable {
+public abstract class BaseMetadataTable implements Table, Serializable {
```

Comment:
Had to make it public so that I can do `instanceof` in Spark.

Reply:
@aokolnychyi, instead of doing this, what about adding a util class in this package that can access the package-private classes? Then we'd be able to move the `SparkUtil.toSerializedTable` into core so it can be shared by Spark and Flink, and we wouldn't need to open up access.

Reply:
How can we move `SparkUtil.toSerializedTable` to core if it depends on `SparkUtil.serializableFileIO`?

Reply:
No longer applies.